br/pkg/stream/client.go

```diff
@@ -0,0 +1,174 @@
+package stream
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/pingcap/errors"
+
+	backuppb "github.com/pingcap/kvproto/pkg/brpb"
+	berrors "github.com/pingcap/tidb/br/pkg/errors"
+	"github.com/pingcap/tidb/br/pkg/redact"
+	"github.com/pingcap/tidb/kv"
+	"go.etcd.io/etcd/clientv3"
+)
+
+// MetaDataClient is the client for operations over metadata.
+type MetaDataClient struct {
+	*clientv3.Client
+}
+
+// PutTask put a task to the metadata storage.
+func (c *MetaDataClient) PutTask(ctx context.Context, task TaskInfo) error {
+	data, err := task.StreamBackupTaskInfo.Marshal()
+	if err != nil {
+		return errors.Annotatef(err, "failed to marshal task %s", task.Name)
+	}
+
+	ops := make([]clientv3.Op, 0, 2+len(task.Ranges))
+	ops = append(ops, clientv3.OpPut(TaskOf(task.GetName()), string(data)))
+	for _, r := range task.Ranges {
+		ops = append(ops, clientv3.OpPut(RangeKeyOf(task.Name, r.StartKey), string(r.EndKey)))
+	}
+	if task.Pausing {
+		ops = append(ops, clientv3.OpPut(Pause(task.Name), ""))
+	}
+
+	txn := c.KV.Txn(ctx)
+	_, err = txn.Then(ops...).Commit()
+	if err != nil {
+		return errors.Annotatef(err, "failed to commit the change for task %s", task.Name)
+	}
+	return nil
+}
+
+// DeleteTask deletes a task, along with its metadata.
+func (c *MetaDataClient) DeleteTask(ctx context.Context, taskName string) error {
+	_, err := c.KV.Txn(ctx).
+		Then(clientv3.OpDelete(TaskOf(taskName)),
+			clientv3.OpDelete(RangesOf(taskName), clientv3.WithPrefix()),
+			clientv3.OpDelete(Pause(taskName))).
+		Commit()
+	if err != nil {
+		return errors.Annotatef(err, "failed to delete task itself %s", taskName)
+	}
+	return nil
+}
+
+func (c *MetaDataClient) PauseTask(ctx context.Context, taskName string) error {
```

Comment:
how we control TiKV pause observe task.🤔

Reply:
According to kenny, pausing / resuming is out of our current scope. 🤔

Reply:
to clarify, it is out of scope because we aren't sure if we want the non-destructive pause/resume or the destructive stop/start, so we delay the feature as it is not essential anyway.